### PR TITLE
Add optional azure-ovn-upgrade jobs for MCO

### PR DESCRIPTION
**Pull request by jluhrsen**
Signed-off-by: Jamo Luhrsen <jluhrsen@gmail.com>

### CODE REVIEW

**Comment by jluhrsen** [comment]
/assign @yuqi-zhang

the rehearsal failures are ok, as the [jobs](https://testgrid.k8s.io/redhat-openshift-ocp-release-4.10-informing#periodic-ci-openshift-release-master-ci-4.10-upgrade-from-stable-4.9-e2e-azure-ovn-upgrade) aren't necessarily very healthy anyway.

we were wanting this while working through a [bz](https://bugzilla.redhat.com/show_bug.cgi?id=2040715) that we hoped would be fixed with a [PR](https://github.com/openshift/machine-config-operator/pull/2948) in MCO. We were having trouble
running jobs with cluster-bot and having an optional presubmit here would have helped. Although, this is now after the fact
(not going to use the MCO PR), it doesn't hurt to have this.

**Comment by deads2k** [comment]
/approve

I'd like to have someone on the MCO team lgtm even an optional job on their repo if they're available.

File: ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml [hunk 1]
```diff
@@ -929,6 +929,79 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-azure,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (yuqi-zhang) [inline ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml:933]: Given that this is a not very health job, are we ok with making this false for now and running on a per-need basis?

I know it's optional but the MCO repo has quite a lot of always_run tests that are very noisy with failures. I'd prefer if we only mark this to always run once we are satisfied with its stability, such that we can actually use it to catch regressions.

Comment (jluhrsen) [reply ci-operator/jobs/openshift/machine-config-operator/openshift-machine-config-operator-master-presubmits.yaml]: yes, no problem. I just updated the PR

**Review by yuqi-zhang — APPROVED** [review]
looks fine to me generally. I see that master/4.10 did both succeed :+1: 

Is `ci/prow/config` supposed to be failing?

**Comment by jluhrsen** [comment]
> Is `ci/prow/config` supposed to be failing?

I can't find this failure? can you point me to it please? I wouldn't expect it to fail though.

**Comment by yuqi-zhang** [comment]
I think it was maybe a flake, sorry, looks like everything is fine now

**Pull request merged**